FirebaseFunctions/Sources/Functions.swift

```diff
@@ -353,8 +446,18 @@ internal enum FunctionsConstants {
       fetcher.setRequestValue(fcmToken, forHTTPHeaderField: Constants.fcmTokenHeader)
     }
 
-    if let appCheckToken = context.appCheckToken {
-      fetcher.setRequestValue(appCheckToken, forHTTPHeaderField: Constants.appCheckTokenHeader)
+    if options?.requireLimitedUseAppCheckTokens == true {
+      if let appCheckToken = context.limitedUseAppCheckToken {
```

Comment:
Does this syntax mean that state is somehow cached or stored in the Functions context? That's undesirable since we plan to make this token very short-lived in the future, and furthermore, every single Functions invocation (with the `requiredLimitedUseAppCheckTokens` options property set) must send a freshly obtained token (which will be one-time-use only) directly from the result of the network call.

Reply:
Good question the _context_ is not cached.

Reply:
A new [context](https://github.com/firebase/firebase-ios-sdk/blob/7534914b6e9b0c8b709e3626b4a911e21c27eaa5/FirebaseFunctions/Sources/Internal/FunctionsContext.swift#L51) is used [per](https://github.com/firebase/firebase-ios-sdk/blob/7534914b6e9b0c8b709e3626b4a911e21c27eaa5/FirebaseFunctions/Sources/Functions.swift#L279) call.

Reply:
Resolving to merge.